plugin/sampling/strategyprovider/adaptive/post_aggregator.go

```diff
@@ -382,24 +382,25 @@ func (p *PostAggregator) isUsingAdaptiveSampling(
 	operation string,
 	throughput serviceOperationThroughput,
 ) bool {
-	if FloatEquals(probability, p.InitialSamplingProbability) {
-		// If the service is seen for the first time, assume it's using adaptive sampling (ie prob == initialProb).
-		// Even if this isn't the case, the next time around this loop, the newly calculated probability will not equal
-		// the initialProb so the logic will fall through.
-		return true
-	}
-	if opThroughput, ok := throughput.get(service, operation); ok {
-		f := TruncateFloat(probability)
-		_, ok := opThroughput.Probabilities[f]
-		return ok
-	}
-	// By this point, we know that there's no recorded throughput for this operation for this round
-	// of calculation. Check the previous bucket to see if this operation was using adaptive sampling
-	// before.
-	if len(p.serviceCache) > 1 {
-		if e := p.serviceCache[1].Get(service, operation); e != nil {
-			return e.UsingAdaptive && !FloatEquals(e.Probability, p.InitialSamplingProbability)
-		}
-	}
-	return false
+	// if FloatEquals(probability, p.InitialSamplingProbability) {
+	// 	// If the service is seen for the first time, assume it's using adaptive sampling (ie prob == initialProb).
+	// 	// Even if this isn't the case, the next time around this loop, the newly calculated probability will not equal
+	// 	// the initialProb so the logic will fall through.
+	// 	return true
+	// }
+	// if opThroughput, ok := throughput.get(service, operation); ok {
+	// 	f := TruncateFloat(probability)
+	// 	_, ok := opThroughput.Probabilities[f]
+	// 	return ok
+	// }
+	// // By this point, we know that there's no recorded throughput for this operation for this round
+	// // of calculation. Check the previous bucket to see if this operation was using adaptive sampling
+	// // before.
+	// if len(p.serviceCache) > 1 {
+	// 	if e := p.serviceCache[1].Get(service, operation); e != nil {
+	// 		return e.UsingAdaptive && !FloatEquals(e.Probability, p.InitialSamplingProbability)
+	// 	}
+	// }
+	// return false
+	return true
```

Comment:
@yurishkuro this causes the unit tests to fail and I believe its messing with the calculations as well. Any ideas on how we can get around this? If we don't hardcode this here however, the probability only gets calculated once.

Reply:
it's very difficult to troubleshoot like this. I would suggest maybe altering tracegen and manually adding the sampler.type=probabilistic / sampler.param=0.5 (any value for now) attributes to the span to see how the system reacts to this. To my knowledge aside from this check the probability used by the sampler should not be affecting the calculations, but I may be wrong.

Reply:
and another thing would help is to expose internal state via expvar so that we can actually monitor how that state changes.